src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -55,6 +54,51 @@ public static Task StopAsync(this IWebHost host, TimeSpan timeout)
         }
 
         /// <summary>
+        /// Block the calling thread until shutdown is triggered via Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        public static void WaitForShutdown(this IWebHost host)
+        {
+            host.WaitForShutdownAsync().GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Block the calling thread until shutdown is triggered via the given token, Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static void WaitForShutdown(this IWebHost host, CancellationToken token)
+        {
+            host.WaitForShutdownAsync(token).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Returns a Task that completes when shutdown is triggered via Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        public static Task WaitForShutdownAsync(this IWebHost host)
+        {
+            return host.WaitForShutdownAsync(CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Returns a Task that completes when shutdown is triggered via the given token, Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static async Task WaitForShutdownAsync(this IWebHost host, CancellationToken token)
```

Comment:
Our APIs usually have a single overload that's async with a `default(CancellationToken)`. We should fix RunAsync as well.